reconcile/ocp_release_ecr_mirror.py

```diff
@@ -0,0 +1,267 @@
+                         f'the mirror. Skipping.')
+                return
+
+            if self.dry_run:
+                return
+
+            subprocess.run(cmd)
+
+    def _is_image_there(self, image):
+        image_obj = Image(image)
+
+        for registry, creds in self.registry_creds['auths'].items():
+            # Getting the credentials for the image_obj
+            registry_obj = urlparse(registry)
+            if registry_obj.netloc != image_obj.registry:
+                continue
+            image_obj.auth = (creds['username'], creds['password'])
+
+            # Checking if the image is already
+            # in the registry
+            if image_obj:
+                return True
+
+        return False
+
+    @staticmethod
+    def _get_cluster_info(cluster):
+        for cluster_info in queries.get_clusters():
+            if cluster_info['name'] != cluster:
+                continue
+            return cluster_info
+
+    @staticmethod
+    def _get_aws_account_info(account):
+        for account_info in queries.get_aws_accounts():
+            if 'name' not in account_info:
+                continue
+            if account_info['name'] != account:
+                continue
+            return account_info
+
+    def _get_ocp_releases(self):
+        ocp_releases = list()
+        clusterimagesets = self.oc_cli.get_all('clusterimageset')
+        for clusterimageset in clusterimagesets['items']:
+            release_image = clusterimageset['spec']['releaseImage']
+            ocp_releases.append(release_image)
+        return ocp_releases
+
+    def _get_quay_creds(self):
+        return self.ocm_cli.get_pull_secrets()
+
+    def _get_ecr_creds(self, account, region):
+        if region is None:
+            region = self.aws_cli.accounts[account]['resourcesDefaultRegion']
+        auth_token = f'{account}/{region}'
+        data = self.aws_cli.auth_tokens[auth_token]
+        auth_data = data['authorizationData'][0]
+        server = auth_data['proxyEndpoint']
+        token = auth_data['authorizationToken']
+        password = base64.b64decode(token).decode('utf-8').split(':')[1]
+
+        return {
+            'auths': {
+                server: {
+                    'username': 'AWS',
+                    'password': password,
+                    'email': 'sd-app-sre@redhat.com',
+                    'auth': token
+                }
+            }
+        }
+
+    def _get_tf_resource_info(self, namespace, identifier):
+        tf_resources = namespace['terraformResources']
+        for tf_resource in tf_resources:
+            if 'identifier' not in tf_resource:
+                continue
+
+            if tf_resource['identifier'] != identifier:
+                continue
+
+            if tf_resource['provider'] != 'ecr':
+                continue
+
+            return {
+                'account': tf_resource['account'],
+                'region': tf_resource.get('region'),
```

Comment:
if `region` is not defined it is decided according to the account's default region. it will probably be too cumbersome here, so this is good enough IMO.